tests/mgr/test_dashboard_url.py

```diff
@@ -0,0 +1,74 @@
+from cli.cephadm.cephadm import CephAdm
+from cli.utilities.utils import get_service_id
+from utility.log import Log
+
+log = Log(__name__)
+
+
+class DashboardUrlError(Exception):
+    pass
+
+
+class CephMgrServiceFailError(Exception):
+    pass
+
+
+class ResourceNotFoundError(Exception):
+    pass
+
+
+def run(ceph_cluster, **kw):
+    """
+    Check if the Ceph dashboard URL consists of host IP and
+    does not bind to host.containers.internal before and after
+    container restart and mgr failover is performed
+    """
+    # Get mgr node
+    node = ceph_cluster.get_nodes(role="mgr")[0]
+
+    # Get dashboard URL from mgr service
+    url = ceph_cluster.get_mgr_services().get("dashboard")
+    if not url:
+        raise ResourceNotFoundError("Dashboard not configured")
+
+    # Check if dashboard URL does not bind to host.containers.internal
+    if "host.containers.internal" in url:
+        raise DashboardUrlError("Dashboard URL binds to host.containers.internal")
+    log.info("Dashboard URL is {}".format(url))
+
+    # Get container id for mgr daemon
+    out, rc = node.exec_command(sudo=True, cmd="podman ps | grep mgr")
+    container_id = out.split(" ")[0]
+
+    # Restart container for mgr daemon
+    out, rc = node.exec_command(sudo=True, cmd=f"podman restart {container_id}")
+
+    # Get dashboard URL from mgr service
+    url = ceph_cluster.get_mgr_services().get("dashboard")
+    if not url:
+        raise ResourceNotFoundError("Dashboard not configured")
```

Comment:
@sayaleeraut why do we have to check the dashboard URL twice?

Reply:
Hey @adityaramteke Do you mean after container restart and mgr fail respectively ??